util.c

```diff
@@ -1550,7 +1554,7 @@ read_file(const char *path, size_t size_hint, char **data, size_t *size)
 	}
 	size_hint = (size_hint < 1024) ? 1024 : size_hint;
 
-	int fd = open(path, O_RDONLY | O_BINARY);
+	int fd = open(path, O_RDONLY | (binary ? O_BINARY : O_TEXT));
```

Comment:
Looks like `binary` is always passed as false, so perhaps just hardcode `O_TEXT` instead?

Also, why do the mingw tests need `O_TEXT` instead of `O_BINARY`?